[2025-08-09T17:03:52.615+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-08-09T17:03:52.653+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_transform_topcv.submit_spark_job manual__2025-08-09T17:03:48.021211+00:00 [queued]>
[2025-08-09T17:03:52.661+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_transform_topcv.submit_spark_job manual__2025-08-09T17:03:48.021211+00:00 [queued]>
[2025-08-09T17:03:52.662+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-08-09T17:03:52.677+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): submit_spark_job> on 2025-08-09 17:03:48.021211+00:00
[2025-08-09T17:03:52.684+0000] {standard_task_runner.py:63} INFO - Started process 1170 to run task
[2025-08-09T17:03:52.688+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'spark_transform_topcv', 'submit_spark_job', 'manual__2025-08-09T17:03:48.021211+00:00', '--job-id', '68', '--raw', '--subdir', 'DAGS_FOLDER/etl_jobs_dag.py', '--cfg-path', '/tmp/tmpp8jh1ei4']
[2025-08-09T17:03:52.694+0000] {standard_task_runner.py:91} INFO - Job 68: Subtask submit_spark_job
[2025-08-09T17:03:52.770+0000] {task_command.py:426} INFO - Running <TaskInstance: spark_transform_topcv.submit_spark_job manual__2025-08-09T17:03:48.021211+00:00 [running]> on host d6c9965fd563
[2025-08-09T17:03:52.907+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_transform_topcv' AIRFLOW_CTX_TASK_ID='submit_spark_job' AIRFLOW_CTX_EXECUTION_DATE='2025-08-09T17:03:48.021211+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-08-09T17:03:48.021211+00:00'
[2025-08-09T17:03:52.909+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-08-09T17:03:52.966+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-08-09T17:03:52.971+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-08-09T17:03:52.972+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 424, in submit
    spark_submit_cmd = self._build_spark_submit_command(application)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 351, in _build_spark_submit_command
    self.log.info("Spark-Submit cmd: %s", self._mask_cmd(connection_cmd))
                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 257, in _mask_cmd
    " ".join(connection_cmd),
    ^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: sequence item 16: expected str instance, list found
[2025-08-09T17:03:52.992+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=spark_transform_topcv, task_id=submit_spark_job, run_id=manual__2025-08-09T17:03:48.021211+00:00, execution_date=20250809T170348, start_date=20250809T170352, end_date=20250809T170352
[2025-08-09T17:03:53.008+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 68 for task submit_spark_job (sequence item 16: expected str instance, list found; 1170)
[2025-08-09T17:03:53.025+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-08-09T17:03:53.042+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-08-09T17:03:53.045+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
